### PR TITLE
Repo cleanup (testing, linters, CI)

### PR DESCRIPTION
**Pull request by cjfinnell**
The main changes can be summarized as:
- upgrade to go1.21
- upgrade linter
- address linter errors
- remove the old/broken integration test that hits AWS/CockroachCloud

### CODE REVIEW

File: integration_test.go [hunk 1]
```diff
@@ -261,9 +261,9 @@ func TestVerifyData(t *testing.T) {
 		aliases = append(aliases, db.image)
 		// Create db and connect
 		_, port, err := createContainer(t, ctx, db.image, db.port, db.env, db.cmd)
-		assert.NoError(t, err)
```
Comment (aronudc) [inline integration_test.go:264]: What's the difference between the assert and require modules? I never really understood it.

File: verify.go [hunk 2]
```diff
@@ -158,7 +152,7 @@ func (c Config) validColumnTarget(columnName string) bool {
 	return false
 }
 
-func (c Config) runTestQueriesOnTarget(ctx context.Context, logger *logrus.Entry, conn *pgx.Conn, schemaTableHashes SingleResult) (SingleResult, error) {
```
Comment (aronudc) [inline verify.go:161]: I'm guessing because the linter flagged that we never return a non-nil error here?

File: .golangci.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ linters:
     - interfacer # deprecated
     - maligned # deprecated
     - scopelint # deprecated
+    - depguard
```
Comment (aronudc) [inline .golangci.yml:13]: What's the issue with this linter?

**Review by aronudc — COMMENTED** [review]
Some Q's

File: integration_test.go (hunk 1 above)
Comment (cjfinnell) [reply integration_test.go:264]: a failed `assert` call will continue the tests, while a failed `require` call will halt (essentially call `t.FailNow()`

File: .golangci.yml (hunk 3 above)
Comment (cjfinnell) [reply .golangci.yml:13]: `depguard` relies on maintaining a whitelist of approved imports - would be useful in a more restrictive development environment, but for us it's just noise for every non-`github.com/udacity` library

File: verify.go (hunk 2 above)
Comment (cjfinnell) [reply verify.go:161]: correct

**Review by aronudc — APPROVED** [review]

**Pull request merged**